### PR TITLE
Move generic argo workflow templates from clinvar-ingest

### PR DESCRIPTION
**Pull request by snwessel**
Moving a few templates out of `clinvar-ingest`:
- ingest-file
- ingest-table
- soft-delete-table

[Here](https://github.com/DataBiosphere/clinvar-ingest/pull/49) is the PR to delete the templates and update the references in `clinvar-ingest`.

### CODE REVIEW

File: charts/argo-templates/templates/ingest-file.yaml [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+        parameters:
+          - name: api-url
+          - name: dataset-id
+          - name: target-path
+          - name: sa-secret
+          - name: sa-secret-key
+      volumes:
+        - name: '{{ $secretVolume }}'
+          secret:
+            secretName: '{{ .Values.repo.accessKey.secretName }}'
```
Comment (danxmoran) [inline charts/argo-templates/templates/ingest-file.yaml]: Should change to use the new Argo input (or else we need a new value for this Helm chart)

File: charts/argo-templates/templates/ingest-file.yaml [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+          - name: dataset-id
+          - name: profile-id
+          - name: source-path
+          - name: target-path
+          - name: sa-secret
+          - name: sa-secret-key
+      volumes:
+        - name: '{{ $secretVolume }}'
+          secret:
+            secretName: '{{ .Values.repo.accessKey.secretName }}'
```
Comment (danxmoran) [inline charts/argo-templates/templates/ingest-file.yaml]: Ditto here

File: charts/argo-templates/templates/ingest-file.yaml [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+        parameters:
+          - name: api-url
+          - name: job-id
+          - name: timeout
+          - name: sa-secret
+          - name: sa-secret-key
+      volumes:
+        - name: '{{ $secretVolume }}'
+          secret:
+            secretName: '{{ .Values.repo.accessKey.secretName }}'
```
Comment (danxmoran) [inline charts/argo-templates/templates/ingest-file.yaml]: And here

File: charts/argo-templates/templates/ingest-file.yaml [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+        image: us.gcr.io/broad-dsp-gcr-public/monster-auth-req-py:1.0.1
+        env:
+          - name: API_URL
+            value: '{{ "{{inputs.parameters.api-url}}" }}'
+          - name: DATASET_ID
+            value: '{{ "{{inputs.parameters.dataset-id}}" }}'
+          - name: TARGET_PATH
+            value: '{{ "{{inputs.parameters.target-path}}" }}'
+          - name: GOOGLE_APPLICATION_CREDENTIALS
+            value: '{{ printf "%s/key.json" $mountPath }}'
```
Comment (danxmoran) [inline charts/argo-templates/templates/ingest-file.yaml]: I think this needs to use the Argo input:
```suggestion
            value: '{{ printf "%s/{{inputs.parameters.sa-secret-key}}" $mountPath }}'
```

File: charts/argo-templates/templates/ingest-file.yaml [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+          - name: DATASET_ID
+            value: '{{ "{{inputs.parameters.dataset-id}}" }}'
+          - name: PROFILE_ID
+            value: '{{ "{{inputs.parameters.profile-id}}" }}'
+          - name: SOURCE_PATH
+            value: '{{ "{{inputs.parameters.source-path}}" }}'
+          - name: TARGET_PATH
+            value: '{{ "{{inputs.parameters.target-path}}" }}'
+          - name: GOOGLE_APPLICATION_CREDENTIALS
+            value: '{{ printf "%s/key.json" $mountPath }}'
```
Comment (danxmoran) [inline charts/argo-templates/templates/ingest-file.yaml]: Same here

File: charts/argo-templates/templates/ingest-file.yaml [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+      inputs:
+        parameters:
+          - name: api-url
+          - name: job-id
+          - name: sa-secret
+          - name: sa-secret-key
+      volumes:
+        - name: '{{ $secretVolume }}'
+          secret:
+            secretName: '{{ .Values.repo.accessKey.secretName }}'
```
Comment (danxmoran) [inline charts/argo-templates/templates/ingest-file.yaml]: Here

File: charts/argo-templates/templates/ingest-file.yaml [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+        image: us.gcr.io/broad-dsp-gcr-public/monster-auth-req-py:1.0.1
+        env:
+          - name: API_URL
+            value: '{{ "{{inputs.parameters.api-url}}" }}'
+          - name: JOB_ID
+            value: '{{ "{{inputs.parameters.job-id}}" }}'
+          - name: TIMEOUT
+            value: '{{ "{{inputs.parameters.timeout}}" }}'
+          - name: GOOGLE_APPLICATION_CREDENTIALS
+            value: '{{ printf "%s/key.json" $mountPath }}'
```
Comment (danxmoran) [inline charts/argo-templates/templates/ingest-file.yaml]: Here

File: charts/argo-templates/templates/ingest-file.yaml [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+        image: us.gcr.io/broad-dsp-gcr-public/monster-auth-req-py:1.0.1
+        env:
+          - name: API_URL
+            value: '{{ "{{inputs.parameters.api-url}}" }}'
+          - name: JOB_ID
+            value: '{{ "{{inputs.parameters.job-id}}" }}'
+          - name: RESULT_FIELD
+            value: fileId
+          - name: GOOGLE_APPLICATION_CREDENTIALS
+            value: '{{ printf "%s/key.json" $mountPath }}'
```
Comment (danxmoran) [inline charts/argo-templates/templates/ingest-file.yaml]: Here

File: charts/argo-templates/templates/ingest-table.yaml [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+          - name: url
+          - name: dataset-id
+          - name: source-path
+          - name: table-name
+          - name: sa-secret
+          - name: sa-secret-key
+      volumes:
+        - name: '{{ $secretVolume }}'
+          secret:
+            secretName: '{{ .Values.repo.accessKey.secretName }}'
```
Comment (danxmoran) [inline charts/argo-templates/templates/ingest-table.yaml]: Here

File: charts/argo-templates/templates/ingest-table.yaml [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+          - name: API_URL
+            value: '{{ "{{inputs.parameters.url}}" }}'
+          - name: DATASET_ID
+            value: '{{ "{{inputs.parameters.dataset-id}}" }}'
+          - name: SOURCE_PATH
+            value: '{{ "{{inputs.parameters.source-path}}" }}'
+          - name: TABLE_NAME
+            value: '{{ "{{inputs.parameters.table-name}}" }}'
+          - name: GOOGLE_APPLICATION_CREDENTIALS
+            value: '{{ printf "%s/key.json" $mountPath }}'
```
Comment (danxmoran) [inline charts/argo-templates/templates/ingest-table.yaml]: Here

File: charts/argo-templates/templates/ingest-table.yaml [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+        parameters:
+          - name: url
+          - name: job-id
+          - name: timeout
+          - name: sa-secret
+          - name: sa-secret-key
+      volumes:
+        - name: '{{ $secretVolume }}'
+          secret:
+            secretName: '{{ .Values.repo.accessKey.secretName }}'
```
Comment (danxmoran) [inline charts/argo-templates/templates/ingest-table.yaml]: Here

File: charts/argo-templates/templates/ingest-table.yaml [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+        image: us.gcr.io/broad-dsp-gcr-public/monster-auth-req-py:1.0.1
+        env:
+          - name: API_URL
+            value: '{{ "{{inputs.parameters.url}}" }}'
+          - name: JOB_ID
+            value: '{{ "{{inputs.parameters.job-id}}" }}'
+          - name: TIMEOUT
+            value: '{{ "{{inputs.parameters.timeout}}" }}'
+          - name: GOOGLE_APPLICATION_CREDENTIALS
+            value: '{{ printf "%s/key.json" $mountPath }}'
```
Comment (danxmoran) [inline charts/argo-templates/templates/ingest-table.yaml]: Here

File: charts/argo-templates/templates/soft-delete-table.yaml [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+          - name: url
+          - name: dataset-id
+          - name: source-path
+          - name: table-name
+          - name: sa-secret
+          - name: sa-secret-key
+      volumes:
+        - name: '{{ $secretVolume }}'
+          secret:
+            secretName: '{{ .Values.repo.accessKey.secretName }}'
```
Comment (danxmoran) [inline charts/argo-templates/templates/soft-delete-table.yaml]: Here

File: charts/argo-templates/templates/soft-delete-table.yaml [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+          - name: API_URL
+            value: '{{ "{{inputs.parameters.url}}" }}'
+          - name: DATASET_ID
+            value: '{{ "{{inputs.parameters.dataset-id}}" }}'
+          - name: SOURCE_PATH
+            value: '{{ "{{inputs.parameters.source-path}}" }}'
+          - name: TABLE_NAME
+            value: '{{ "{{inputs.parameters.table-name}}" }}'
+          - name: GOOGLE_APPLICATION_CREDENTIALS
+            value: '{{ printf "%s/key.json" $mountPath }}'
```
Comment (danxmoran) [inline charts/argo-templates/templates/soft-delete-table.yaml]: Here

File: charts/argo-templates/templates/soft-delete-table.yaml [hunk 15]
```diff
@@ -0,0 +1,122 @@
...
+        parameters:
+          - name: url
+          - name: job-id
+          - name: timeout
+          - name: sa-secret
+          - name: sa-secret-key
+      volumes:
+        - name: '{{ $secretVolume }}'
+          secret:
+            secretName: '{{ .Values.repo.accessKey.secretName }}'
```
Comment (danxmoran) [inline charts/argo-templates/templates/soft-delete-table.yaml]: Here

File: charts/argo-templates/templates/soft-delete-table.yaml [hunk 16]
```diff
@@ -0,0 +1,122 @@
...
+        image: us.gcr.io/broad-dsp-gcr-public/monster-auth-req-py:1.0.1
+        env:
+          - name: API_URL
+            value: '{{ "{{inputs.parameters.url}}" }}'
+          - name: JOB_ID
+            value: '{{ "{{inputs.parameters.job-id}}" }}'
+          - name: TIMEOUT
+            value: '{{ "{{inputs.parameters.timeout}}" }}'
+          - name: GOOGLE_APPLICATION_CREDENTIALS
+            value: '{{ printf "%s/key.json" $mountPath }}'
```
Comment (danxmoran) [inline charts/argo-templates/templates/soft-delete-table.yaml]: Here

**Review by danxmoran — COMMENTED** [review]
Sorry for repetitive comments, wanted to make sure everything was covered.

The biggest thing missing from this diff are the scripts themselves. Every file referenced in a `.Files.Lines` call needs to be moved from ClinVar's `scripts/` dir into this chart's `scripts/` dir.

File: charts/argo-templates/templates/ingest-file.yaml (hunk 1 above)
Comment (snwessel) [reply charts/argo-templates/templates/ingest-file.yaml]: It looks like one of the existing input parameters (`sa-secret`) actually has this value, so I switched to using that.

**Review by quazi-h — APPROVED** [review]
Looks good to me.

File: charts/argo-templates/templates/soft-delete-table.yaml [hunk 17]
```diff
@@ -0,0 +1,122 @@
...
+          - name: TIMEOUT
+            value: '{{ "{{inputs.parameters.timeout}}" }}'
+          - name: GOOGLE_APPLICATION_CREDENTIALS
+            value: '{{ printf "%s/{{inputs.parameters.sa-secret-key}}" $mountPath }}'
+        volumeMounts:
+          - name: '{{ $secretVolume }}'
+            mountPath: '{{ $mountPath }}'
+        command: [python]
+        source: |
+          {{- range .Files.Lines "scripts/poll-ingest-job.py" }}
```
Comment (danxmoran) [inline charts/argo-templates/templates/soft-delete-table.yaml]: Nitpick: There's now a helper that could make this slightly less verbose, you could do:
```
source: |
{{- include "argo.render-lines" (.Files.Lines "scripts/poll-ingest-job.py") | indent 10 }}
```
Can be used for all the script sources.

**Review by danxmoran — COMMENTED** [review]
Last piece: The version in `argo-templates/Chart.yaml` needs incremented for the updates to be published on merge.

File: charts/argo-templates/templates/soft-delete-table.yaml (hunk 17 above)
Comment (snwessel) [reply charts/argo-templates/templates/soft-delete-table.yaml]: Thank you! I just updated with those changes. 

Also, a quick question on testing: is there a way to test any of these changes before this branch is merged? Or is it just a matter of merging this, and then seeing if my changes to the clinvar orchestration can be deployed without errors?

Comment (danxmoran) [reply charts/argo-templates/templates/soft-delete-table.yaml]: There's no clean way to test this without merging and pulling the auto-deploy. It's super annoying, but I'm hoping it won't bite us that often in practice once we stop with these big shifts of functionality into monster-helm

**Review by danxmoran — APPROVED** [review]

**Pull request merged**